camayoc/tests/qcs/test_scan.py

```diff
@@ -0,0 +1,185 @@
+# coding=utf-8
+"""Tests for ``Network Profile`` API endpoint for quipucords server.
+
+:caseautomation: automated
+:casecomponent: api
+:caseimportance: high
+:caselevel: integration
+:requirement: Sonar
+:testtype: functional
+:upstream: yes
+"""
+
+import logging
+import pytest
+import time
+
+from camayoc import config
+from camayoc.exceptions import ConfigFileNotFoundError
+from camayoc.qcs_models import (
+    HostCredential,
+    NetworkProfile,
+    Scan,
+)
+
+
+def profiles():
+    """Gather profiles from config file."""
+    try:
+        profs = config.get_config()['qcs']['profiles']
+    except ConfigFileNotFoundError:
+        profs = []
+    return profs
+
+
+LOGGER = logging.Logger(name='camayoc')
+
+
+def wait_until_complete(scan, timeout=120):
+    """Wait until the scan has either failed or completed.
+
+    This method should not be called on scan jobs that have not yet been
+    created, are paused, or are canceled.
+
+    The default timeout is set to 120 seconds, but can be overridden if it is
+    anticipated that a scan may take longer to complete.
+    """
+    if scan.status and scan.status() in ['paused', 'cancelled']:
+        LOGGER.critical(msg=('Camayoc Warning: Waiting for a job that will not'
+                             ' start because it is paused or canceled'))
```

Comment:
Again, I could raise an exception here instead depending on what you think is best.